docs/development/tests.md

```diff
@@ -18,10 +18,62 @@ Integration tests include the basic working of:
 - **data validation**: corrupted etcd data should be marked for deletion and restoration should be triggered
 - **restoration**: etcd data should be restored correctly from latest set of snapshots (full + deltas)
 
+**Note**: The tests expects that the aws credentials are present in the `$HOME/.aws` directory. Make sure to provide the correct credentials before running the tests.
+
+These tests can be run locally with both `etcd` and `etcdbr` running as processes.  To execute the tests, run the following command:
+
+```sh
+make integration-test
+```
+
 ### Unit tests
 
 Each package within this repo contains its own set of unit tests to test the functionality of the methods contained within the packages.
 
 ### Performance regression tests
 
 These tests help check any regression in performance in terms of memory consumption and CPU utilization.
+
+### End-to-end tests
+
+The e2e tests for etcd-backup-restore are cluster based tests located in the `test/e2e` package. These tests are run on a Kubernetes cluster and test the full functionality of etcd-backup-restore. The tests create a provider namespace on the cluster and deploy the [etcd-backup-restore helm chart](../../chart/etcd-backup-restore) which in turn deploys the required secrets, configmap, services and finally the statefulset which deploys the pod that runs etcd and backup-restore as a sidecar.
+
+These tests are setup to be run with both emulators and real cloud providers. The emulators can be used for local development and testing as well as running jobs to test code changes when a PR is raised. The real cloud providers can be used for testing in a real cloud environment to ensure that the changes work as expected in an actual environment.
+
+Currently, the tests can be run using the following cloud providers:
+
+- AWS
+- GCP
+- Azure
+
+#### Running the e2e tests with the emulators
+
+To run the e2e tests with the emulators, run the following command:
+
+```sh
+make ci-e2e-kind PROVIDERS="{providers}"
```

Comment:
Please update to `make PROVIDERS="{providers}" ci-e2e-kind` as previously mentioned.